test/src/test/java/org/corfudb/runtime/clients/TestClientRouter.java

```diff
@@ -318,4 +509,18 @@ public CorfuMsg simulateSerialization(CorfuMsg message) {
         return msg;
     }
 
+    public RequestMsg simulateSerialization(RequestMsg message) throws IOException {
+        /* simulate serialization/deserialization */
+        ByteBuf oBuf = Unpooled.buffer();
+        try (ByteBufOutputStream requestOutputStream = new ByteBufOutputStream(oBuf)) {
+            message.writeTo(requestOutputStream);
+            oBuf.resetReaderIndex();
+            try (ByteBufInputStream msgInputStream = new ByteBufInputStream(oBuf)) {
+                RequestMsg msg = RequestMsg.parseFrom(msgInputStream);
+                oBuf.release();
```

Comment:
the release of the buffer should probably be in a `finally` block and not within the try itself.

Reply:
Good catch -- Done.